### PR TITLE
Regenerate files from latest templates.

### PR DESCRIPTION
**Pull request by Zimmi48**
Includes a move from the Gitter badge to the Zulip badge in README.

### CODE REVIEW

**Comment by Zimmi48** [comment]
@herbelin 
CI fails because `master` does not build anymore with Coq dev. In fact, it hasn't been building for a long time. The `v8.10` branch is ahead and fix compilation with 8.10. If this project is unmaintained, we should mark it as is and as looking for a new maintainer.

**Comment by herbelin** [comment]
Should I merge anyway?

If someone is willing to be a maintainer, I'll gladly accept. (And otherwise, I can continue upgrading it.)

**Comment by Zimmi48** [comment]
Yes 👍

**Comment by herbelin** [comment]
@Zimmi48: I updated Stalmarck to 8.11 and 8.12 but:
- This now includes the changes made by @palmskog to the `yml` and `README.md` files.
- Your PR is not compatible anymore: should I just overwrite the files?
- What are the coq-community procedures to create releases and opam packages? (If possible in batch mode.)

**Comment by Zimmi48** [comment]
I'll rebase the PR.

**Comment by palmskog** [comment]
> What are the coq-community procedures to create releases and opam packages? (If possible in batch mode.)

Unfortunately we are still doing this manually, i.e., one goes to https://github.com/coq-community/stalmarck/releases and uses the "Draft a new release". Then, the opam file has to be modified with the release tarball/checksum before submission to the Coq opam archive.

**Comment by herbelin** [comment]
For coq-contribs, putting a tag creates automatically a release iirc and a script prepares the opam package automatically. Maybe could I simply add my `make-opam` script in the stalmarck repository (unless you would prefer to have such script available in a public place).

**Comment by palmskog** [comment]
> Maybe could I simply add my `make-opam` script in the stalmarck repository (unless you would prefer to have such script available in a public place).

Per earlier discussions with Théo, I believe we have consensus that until there is a smooth general solution for release automation, maintainers are encouraged to use whatever automation works best for them. However, note that we prefer "full" releases on GitHub rather than just tags.

**Comment by herbelin** [comment]
> However, note that we prefer "full" releases on GitHub rather just tags.

I see. So GitHub does not support driving the "Drafting a new release" by sending batch requests?

**Comment by Zimmi48** [comment]
I don't know what you mean by "batch" requests. But you can certainly automate release creation by interacting with Coq's API.

**Comment by Zimmi48** [comment]
I wonder what's the use of CI if it's not to take its results into account. You included commit 64610c6479c17b1eb28fd82ac7dee7ec43cfb2ec in `master`, `v8.11`, `v8.12` and `v8.11.0` despite this commit making the README and the CI say that this project is compatible with Coq 8.10 (and thus making the CI fail for these branches and tags). I'm reverting this as part of this PR.

**Comment by Zimmi48** [comment]
Actually I was wrong, you did update manually the version number present in the README and CI (although only to 8.11 even for the master and v8.12 branches) and the opam file up to 8.12. Modifying only the `meta.yml` file and regenerating the rest from the templates would avoid any risks of inconsistencies.

File: meta.yml [hunk 1]
```diff
@@ -3,16 +3,19 @@ fullname: Stalmarck
...
+travis: true
 
 synopsis: Proof of Stålmarck's algorithm in Coq
 
 description: |
   A two-level approach to prove tautologies using Stålmarck's algorithm in Coq.
 
+doi: 10.1007/3-540-44659-1_24
+
 paper:
```
Comment (palmskog) [inline meta.yml]:
```suggestion
publications:
```
For the paper to show up in the README this needs to be changed (we initially assumed just one paper).

File: meta.yml [hunk 2]
```diff
@@ -3,16 +3,19 @@ fullname: Stalmarck
...
-  doi: 10.1007/3-540-44659-1_24
-  url: https://www.irif.fr/~letouzey/download/stalmarck.ps.gz
-  title: Formalizing Stålmarck's Algorithm in Coq
+  pub_doi: 10.1007/3-540-44659-1_24
```
Comment (palmskog) [inline meta.yml]:
```suggestion
- pub_doi: 10.1007/3-540-44659-1_24
```
This needs to be a list item now.

**Comment by Zimmi48** [comment]
Thanks for noticing. I was aware of this but apparently didn't perform the change in full (only renamed the internal fields).

**Comment by Zimmi48** [comment]
Looks like CI is still failing with Coq master...

**Comment by herbelin** [comment]
> Modifying only the meta.yml file and regenerating the rest from the templates would avoid any risks of inconsistencies.

Sorry about my lack of familiarity with the `meta.yml` file. What means "regenerating the rest from the templates" (i.e. what is the "rest" and how to regenerate it?). 

More generally, what kind of modifications are supposed to be done to the `meta.yml` file? For instance, when to change the `text` and `opam` item at `supported_coq_versions`, or the `version_or_url` at `tested_coq_nix_versions`, etc. Also, how to know the role of each the field, and where is the script which interprets them?

> Looks like CI is still failing with Coq master...

Was a stupid reason (I committed a change too early). Should now be ok.

> I don't know what you mean by "batch" requests. 

It now almost automatic. There is a script `make-release opamdir releasenumber` which tag a branch, push the tag to github, and produce the corresponding opam package in the opam local directory. There is no proper release item as if using "the draft a new release" button though. This is the latter which I don't know it can be driven by a script.

**Comment by herbelin** [comment]
I found the answer to some of my questions in the template directory (which probably I have been aware of at some time and forgot). My remaining questions are:
- What is supposed to be given in the field "Compatible Coq versions:" of the `README`? What about indicating an (extensible) list of all supported versions and to add the date at which the affirmation is made, thus, this would be a correct information in the `README` of all branches. What do you think?
- Apparently, each branch with a `.travis.yml` file is tested, so I conclude that the configuration file should be specific to each branch and tell to test the corresponding branch with either nix or opam. So what was done seems correct. I don't know if it is worth that the opam packages are tested too, in addition to the nix ones. Also, your PR will update the master branch, which is nice.

A rebase is needed.

**Comment by Zimmi48** [comment]
>What is supposed to be given in the field "Compatible Coq versions:" of the `README`? What about indicating an (extensible) list of all supported versions and to add the date at which the affirmation is made, thus, this would be a correct information in the `README` of all branches. What do you think?

It could also work. Up to now, we have rather made the README explain what was in the corresponding branch (said otherwise: state compatibility with master and redirect to other branches and tags for compatibility with other versions).

I'm not sure what is the second question.

Looking into the rebase.

**Comment by palmskog** [comment]
This looks good to me, thanks @Zimmi48. I can take it from here.

**Comment by herbelin** [comment]
> I'm not sure what is the second question.

Not really a question. It was somehow about what is the policy to test either nix or opam, and, in the case of stalmarck, why nix and not opam. Maybe not so important...

@palmskog: thanks for taking over.

**Comment by Zimmi48** [comment]
OK, thanks for clarifying. There isn't really a policy. I suppose the reason why nix and not opam in the case of stalmarck is simply because stalmarck got its CI early at a time when CI with opam was not supported in the template. See also [this comparison](https://github.com/coq-community/manifesto/wiki/Continuous-Integration-with-Nix#alternative-setup-and-comparison).

**Pull request merged**